src/OSPSuite.Core/Domain/Services/SimulationResultsToDataTableConverter.cs

```diff
@@ -0,0 +1,138 @@
+﻿using System.Data;
+using System.Linq;
+using System.Threading.Tasks;
+using OSPSuite.Core.Domain.Data;
+using OSPSuite.Core.Domain.UnitSystem;
+using OSPSuite.Core.Services;
+using OSPSuite.Utility.Collections;
+using OSPSuite.Utility.Extensions;
+using static OSPSuite.Core.Domain.Constants.SimulationResults;
+
+namespace OSPSuite.Core.Domain.Services
+{
+   public interface ISimulationResultsToDataTableConverter
+   {
+      /// <summary>
+      ///    Creates a <see cref="DataTable" /> containing  the results of the simulation.
+      /// </summary>
+      /// <remarks>The format of the table will be one column per output and one row per individual</remarks>
+      /// <param name="simulation">Simulation used to create the results</param>
+      /// <param name="simulationResults">Actual results to convert to <see cref="DataTable" /></param>
+      Task<DataTable> ResultsToDataTable(SimulationResults simulationResults, IModelCoreSimulation simulation);
+
+      /// <summary>
+      ///    Creates a <see cref="DataTable" /> containing the PK-Analyses of the populationSimulation.
+      /// </summary>
+      /// <remarks>The format of the table will be IndividualId, QuantityPath, ParameterName, Value, Unit</remarks>
+      /// <param name="simulation">Simulation used to calculate the PK-Analyses</param>
+      /// <param name="pkAnalyses">Actual pkAnalyses to convert to <see cref="DataTable" /></param>
+      Task<DataTable> PKAnalysesToDataTable(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation);
+   }
+
+   public class SimulationResultsToDataTableConverter : ISimulationResultsToDataTableConverter
+   {
+      private readonly IEntitiesInSimulationRetriever _quantityRetriever;
+      private readonly IDisplayUnitRetriever _displayUnitRetriever;
+      private readonly IDimension _timeDimension;
+
+      public SimulationResultsToDataTableConverter(IDimensionFactory dimensionFactory, IEntitiesInSimulationRetriever quantityRetriever, IDisplayUnitRetriever displayUnitRetriever)
+      {
+         _quantityRetriever = quantityRetriever;
+         _displayUnitRetriever = displayUnitRetriever;
+         _timeDimension = dimensionFactory.Dimension(Constants.Dimension.TIME);
+      }
+
+      public Task<DataTable> ResultsToDataTable(SimulationResults simulationResults, IModelCoreSimulation simulation)
+      {
+         return Task.Run(() => createResultDataToExport(simulationResults, simulation));
+      }
+
+      public Task<DataTable> PKAnalysesToDataTable(PopulationSimulationPKAnalyses pkAnalyses, IModelCoreSimulation simulation)
+      {
+         return Task.Run(() => createPKAnalysesDataToExport(pkAnalyses, simulation));
+      }
+
+      private DataTable createResultDataToExport(SimulationResults simulationResults, IModelCoreSimulation simulation)
```

Comment:
THis comes from PKSim directly. This is not ideal because the results are then twice in memory. For big export to CSV, this could create contention. However this works for now, especially with 64bit